src/GafferImage/Empty.cpp

```diff
@@ -0,0 +1,161 @@
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/Context.h"
+
+#include "GafferImage/Empty.h"
+#include "GafferImage/ImageAlgo.h"
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace GafferImage;
+using namespace Gaffer;
+
+//////////////////////////////////////////////////////////////////////////
+// Empty implementation
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( Empty );
+
+size_t Empty::g_firstPlugIndex = 0;
+
+Empty::Empty( const std::string &name )
+	: ImageNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new FormatPlug( "format" ) );
+}
+
+Empty::~Empty()
+{
+}
+
+GafferImage::FormatPlug *Empty::formatPlug()
+{
+	return getChild<FormatPlug>( g_firstPlugIndex );
+}
+
+const GafferImage::FormatPlug *Empty::formatPlug() const
+{
+	return getChild<FormatPlug>( g_firstPlugIndex );
+}
+
+void Empty::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ImageNode::affects( input, outputs );
+
+	if( formatPlug()->displayWindowPlug()->isAncestorOf( input ) )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+		outputs.push_back( outPlug()->dataWindowPlug() );
+	}
+	else if( input == formatPlug()->pixelAspectPlug() )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+	}
+}
+
+void Empty::hashFormat( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashFormat( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+GafferImage::Format Empty::computeFormat( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue();
+}
+
+void Empty::hashDataWindow( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashDataWindow( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+Imath::Box2i Empty::computeDataWindow( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue().getDisplayWindow();
+}
+
+IECore::ConstCompoundDataPtr Empty::computeMetadata( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return outPlug()->metadataPlug()->defaultValue();
+}
+
+bool Empty::computeDeep( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return true;
+}
+
+void Empty::hashSampleOffsets( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImagePlug::emptyTileSampleOffsets()->hash( h );
```

Comment:
I _think_ this might work out OK in practice, but it breaks the general rule "either call base class and then append, or assign direct". So I think this should be `h = ImagePlug::emptyTileSampleOffsets()->hash( h )`.